railgun/manager/run.go

```diff
@@ -338,6 +343,7 @@ func Run(ctx context.Context, c *config.Config) error {
 	} else {
 		setupLog.Info(`ingresses.networking.internal.knative.dev v1alpha1 CRD not available on cluster.
 		Disabling Knative controller`)
+		c.KnativeIngressEnabled = util.EnablementStatusDisabled
```

Comment:
The effect of this change is that we'll change the users intention which I don't think is something we want to do. Can you help me to understand what problem this fixes, or what the intention behind this change was?

Reply:
As the log indicates "disabling Knative controller", and the knative controller is disabled, so the flag needs be set to "util.EnablementStatusDisabled".

Reply:
Including `controllers = append(controllers, controller)` in the clause above means that we won't load it into list we [later load into the manager](https://github.com/Kong/kubernetes-ingress-controller/blob/e0292952de52a68ef1218c45d959a8fc98745f8a/railgun/manager/run.go#L347-L351). If it's not in that list, the config does nothing.

I'd suggest removing it out of principle for not modifying the user config struct, but leaving it in place is benign.